policy/release/olm/olm.rego

```diff
@@ -194,20 +223,7 @@ deny contains result if {
 deny contains result if {
 	_release_restrictions_apply
 
-	snapshot_components := input.snapshot.components
-	component_images_digests := [component_image.digest |
-		some component in snapshot_components
-		component_image := image.parse(component.containerImage)
-	]
-
-	some related_images in _related_images(input.image)
-
-	unmatched_image_refs := [related |
-		some related in related_images
-		not related.digest in component_images_digests
-	]
-
-	some unmatched_image in unmatched_image_refs
+	some unmatched_image in _related_images_not_in_snapshot
 	unmatched_ref := sprintf("%s@%s", [unmatched_image.repo, unmatched_image.digest])
```

Comment:
```suggestion
	unmatched_ref := unmatched_image.repo
	unmatched_ref := sprintf("%s@%s", [unmatched_image.repo, unmatched_image.digest]) with unmatched_image.digest != ""
	unmatched_ref := sprintf("%s:%s", [unmatched_image.repo, unmatched_image.tag]) with unmatched_image.digest == ""; unmatched_image.tag != ""
```

This would be nice-to-have: Ideally, I'd want to do something like the above ^ so that we pass valid image refs to the `ec.oci.descriptor()` function.

But first of all, this isn't a valid rego code, and also this isn't necessary because we'd anyway be getting a nice error message in the logs to point us to an invalid image ref.

Reply:
Okay yeah, it makes sense.